trie/zk_trie_impl_test.go

```diff
@@ -9,26 +10,23 @@ import (
 	zkt "github.com/scroll-tech/zktrie/types"
 )
 
-var lcEff *big.Int
-var prime = int64(5915587277)
-
-func dummyHash(arr []*big.Int) (*big.Int, error) {
-
-	sum := big.NewInt(0)
-
-	for _, bi := range arr {
-		nbi := big.NewInt(0).Mul(bi, bi)
-		sum = sum.Mul(sum, sum)
-		sum = sum.Mul(sum, lcEff)
-		sum = sum.Add(sum, nbi)
+func init() {
+	var lcEff *big.Int
+	var prime = int64(5915587277)
+
+	testHash := func(arr []*big.Int) (*big.Int, error) {
+		sum := big.NewInt(0)
+		for _, bi := range arr {
+			nbi := big.NewInt(0).Mul(bi, bi)
+			sum = sum.Mul(sum, sum)
+			sum = sum.Mul(sum, lcEff)
+			sum = sum.Add(sum, nbi)
+		}
+		return sum.Mod(sum, big.NewInt(prime)), nil
 	}
-	return sum.Mod(sum, big.NewInt(prime)), nil
-}
 
-func init() {
```

Comment:
interesting. `init()` in `trie/zk_trie_impl_test.go` exists before, why it runs correctly (init the hash function multiple times, but `setHashScheme` is `sync.Once`)? @noel2004

Reply:
I see. `init()` of a dependent package will be called later. Thus, scroll geth zktrie `init()` first (`poseidon.HashFixed`), then zktrie `init()`.

Reply:
we need a dummy hash to make most test run correctly. I think `init()` is called while package is imported so it would not execute mutiple times. Of course we can use `TestMain` instead, which is a more formal way.

Reply:
**RE:** I think init() is called while package is imported so it would not execute multiple times.
But init() in different places would try to overwrite the hash function. e.g., [poseidon.HashFixed](https://github.com/scroll-tech/go-ethereum/blob/scroll-v3.3.0/trie/zk_trie.go#L41) and [dummyHash](https://github.com/scroll-tech/zktrie/blob/v0.5.3/trie/zk_trie_impl_test.go#L31).

Reply:
btw: everything in "_test.go" files is not visible in building. So put `init()` in test file has no effect on the released code.

Not a pretty way but it just work.

Reply:
I see then.